### PR TITLE
Common method that verify groups and workspace access

### PR DESCRIPTION
**Pull request by aweng98**
[rawls PR 1072](https://github.com/broadinstitute/rawls/pull/1072), [PR 1070](https://github.com/broadinstitute/rawls/pull/1070) and [Sam PR 294](https://github.com/broadinstitute/sam/pull/294) depend on this PR.

### CODE REVIEW

File: serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/service/Rawls.scala [hunk 1]
```diff
@@ -59,25 +57,6 @@ trait Rawls extends RestClient with LazyLogging {
       deleteRequest(s"${url}api/billing/$projectName/googleRole/$googleRole/$email")
     }
 
-    private def waitUntilBillingProjectIsReady(projectName: String, billingAccount: String)(implicit token: AuthToken): Unit = {
```
Comment (aweng98) [inline serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/service/Rawls.scala:62]: removed this in workbench-lib. the waiting and retrying of new billing project is moved to Rawls BillingSpiSpec. See [Rawls PR 1072](https://github.com/broadinstitute/rawls/pull/1072)

File: serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/service/test/AuthDomainMatcher.scala [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+  val rawlsErrorMsg = "does not exist"
+
+  /**
+    * user can see the workspace in workspaces list and
+    *  can see workspace detail and see expected auth-domains
+    */
+  def checkVisibleAndAccessible(user: Credentials, projectName: String, workspaceName: String, authDomains: List[String]): Unit = {
+
+    val allWorkspacenames: Seq[String] = Rawls.workspaces.getWorkspaceNames()(user.makeAuthToken())
```
Comment (marctalbott) [inline serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/service/test/AuthDomainMatcher.scala]: nit: capitalize names here `allWorkspaceNames` and in the later functions `workspaceNames`

File: serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/service/test/AuthDomainMatcher.scala [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  //  NotVisible -> Not found in workspace list
+  //  NotAccessible -> Not authorized to see workspace detail
+
+  val rawlsErrorMsg = "does not exist"
+
+  /**
+    * user can see the workspace in workspaces list and
+    *  can see workspace detail and see expected auth-domains
+    */
+  def checkVisibleAndAccessible(user: Credentials, projectName: String, workspaceName: String, authDomains: List[String]): Unit = {
```
Comment (marctalbott) [inline serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/service/test/AuthDomainMatcher.scala]: Could you just pass in the user's auth token as an implicit?

Comment (aweng98) [reply serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/service/test/AuthDomainMatcher.scala]: 👍 yes

File: serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/service/test/AuthDomainMatcher.scala (hunk 2 above)
Comment (aweng98) [reply serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/service/test/AuthDomainMatcher.scala]: 👍

**Review by marctalbott — APPROVED** [review]

**Review by ansingh7115 — APPROVED** [review]

**Pull request merged**